src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -2735,4 +2938,14 @@ private static ClientCnxnSocket getClientCnxnSocket() throws IOException {
             throw ioe;
         }
     }
+
+    /**
+     * VisibleForTesting
+     * @return ClientCnxn
+     */
+    public ClientCnxn getCnxn() {
```

Comment:
Is it possible to not make this method public? It is odd if you say are on an IDE looking at the available methods of the handle and see this.